cgi-bin/LJ/User/Account.pm

```diff
@@ -1282,7 +1282,8 @@ use Carp;
 sub canonical_username {
     my $input = lc( $_[0] // '' );
     my $user = "";
-    if ( $input =~ /^\s*([a-z0-9_\-]{1,25})\s*$/ ) {  # good username
+    my $exp = '^\s*([a-z0-9_\-]{1,' . $LJ::USERNAME_MAXLENGTH . '})\s*$';
```

Comment:
This still confuses me, not sure what this line does :)

I'll delete it to get this merged..